src/quickgui/qgsquickidentifykit.cpp

```diff
@@ -0,0 +1,259 @@
+      if ( vl )
+      {
+        QgsFeatureList featureList = identifyVectorLayer( vl, mapPoint );
+
+        for ( const QgsFeature &feature : featureList )
+        {
+          results.append( QgsQuickFeatureLayerPair( feature, vl ) );
+        }
+      }
+    }
+
+    QgsDebugMsg( QStringLiteral( "IdentifyKit identified %1 results" ).arg( results.count() ) );
+  }
+
+  return results;
+}
+
+static QgsQuickFeatureLayerPair _closestFeature( const QgsQuickFeatureLayerPairs &results, const QgsMapSettings &mapSettings, const QPointF &point )
+{
+  QgsPointXY mapPoint = mapSettings.mapToPixel().toMapCoordinates( point.toPoint() );
+  QgsGeometry mapPointGeom( QgsGeometry::fromPointXY( mapPoint ) );
+
+  double distMin = 1e10;
+  int iMin = -1;
+  for ( int i = 0; i < results.count(); ++i )
+  {
+    const QgsQuickFeatureLayerPair &res = results.at( i );
+    QgsGeometry geom( res.feature().geometry() );
+    try
+    {
+      geom.transform( mapSettings.layerTransform( res.layer() ) );
+    }
+    catch ( QgsCsException &e )
+    {
+      Q_UNUSED( e );
+      // Caught an error in transform
+      continue;
+    }
+
+    double dist = geom.distance( mapPointGeom );
+    if ( dist < distMin )
+    {
+      iMin = i;
+      distMin = dist;
+    }
+  }
+
+  if ( results.empty() )
+  {
+    return QgsQuickFeatureLayerPair();
+  }
+  else
+  {
+    return results.at( iMin );
+  }
+}
+
+QgsQuickFeatureLayerPair QgsQuickIdentifyKit::identifyOne( const QPointF &point, QgsVectorLayer *layer )
+{
+  QgsQuickFeatureLayerPairs results = identify( point, layer );
+  return _closestFeature( results, mMapSettings->mapSettings(), point );
+}
+
+QgsFeatureList QgsQuickIdentifyKit::identifyVectorLayer( QgsVectorLayer *layer, const QgsPointXY &point ) const
+{
+  QgsFeatureList results;
+
+  if ( !layer || !layer->isSpatial() )
+    return results;
+
+  if ( !layer->isInScaleRange( mMapSettings->mapSettings().scale() ) )
+    return results;
+
+  QgsFeatureList featureList;
+
+  // toLayerCoordinates will throw an exception for an 'invalid' point.
+  // For example, if you project a world map onto a globe using EPSG 2163
+  // and then click somewhere off the globe, an exception will be thrown.
+  try
+  {
+    // create the search rectangle
+    double searchRadius = searchRadiusMU();
+
+    QgsRectangle r;
+    r.setXMinimum( point.x() - searchRadius );
+    r.setXMaximum( point.x() + searchRadius );
+    r.setYMinimum( point.y() - searchRadius );
+    r.setYMaximum( point.y() + searchRadius );
+
+    r = toLayerCoordinates( layer, r );
+
+    QgsFeatureRequest req;
+    req.setFilterRect( r );
+    req.setLimit( mFeaturesLimit );
+    req.setFlags( QgsFeatureRequest::ExactIntersect );
+
+    QgsFeatureIterator fit = layer->getFeatures( req );
+    QgsFeature f;
+    while ( fit.nextFeature( f ) )
+      featureList << QgsFeature( f );
+  }
+  catch ( QgsCsException &cse )
+  {
+    QgsDebugMsg( tr( "Invalid point and proceed with no features found." ) );
```

Comment:
If something deserves translation it should probably be sent to QgsMessageLog (or translation removed)